### PR TITLE
Refactor to specs

### PR DESCRIPTION
**Pull request by javierbustillo**

### CODE REVIEW

File: DAO/MessageDAO.py [hunk 1]
```diff
@@ -125,8 +125,9 @@ def get_num_like_photos_daily(self, pid, date, like):
     def get_trending_hashtags_day(self, date):
         cursor = self.get_cursor()
         end_date = date + relativedelta(days=1)
-        query = "select count(*) as num, hid, hashtag from hashtags_messages natural inner join messages " \
+        query = "with trending as (select count(*) as num, hid from hashtags_messages natural inner join messages " \
                 "natural inner join hashtags where messages.created_on > %s and messages.created_on < %s " \
-                "group by hid order by num desc"
+                "group by hid order by num desc limit 10)" \
+                "select hashtag from hashtags inner join trending on trending.hid = hashtags.hid"
```
Comment (ChristianPerez34) [inline DAO/MessageDAO.py]: Does using a NATURAL INNER JOIN here not give you the same results?

File: Handlers/Message.py [hunk 2]
```diff
@@ -33,73 +33,76 @@ def dislike_message(self, mid):
...
+        trending_hashtags = []
         for i in range(7):
             day_to_get = today - relativedelta(days=i)
             hashtags = self.dao.get_trending_hashtags_day(day_to_get)
-            hashtags.append({'%s' % day_to_get: hashtags})
-        return jsonify(result=hashtags)
+            hashtag_day = []
+            for j, hashtag in enumerate(hashtags):
+                hashtag_day.append({'hashtag': hashtag['hashtag'], 'position': j+1})
+            trending_hashtags.append({'day': day_to_get, 'hashtags': hashtag_day}.copy())
```
Comment (ChristianPerez34) [inline Handlers/Message.py]: Why a copy here?

Comment (javierbustillo) [reply Handlers/Message.py]: Yeah copy se va, es viejo

File: DAO/MessageDAO.py (hunk 1 above)
Comment (javierbustillo) [reply DAO/MessageDAO.py]: surprisingly no... It wasn't working when doing it that way. I'm thinking it doesn't work with subqueries but it doesn't make sense

File: SQL Scripts/CreateDB.sql [hunk 3]
```diff
@@ -31,7 +31,7 @@ CREATE TABLE Messages (
 );
 
 CREATE TABLE Photo (
-    image BYTEA NOT NULL,
+    image varchar(100) NOT NULL,
```
Comment (ChristianPerez34) [inline SQL Scripts/CreateDB.sql]: [Best database field type for a url](https://stackoverflow.com/questions/219569/best-database-field-type-for-a-url)

Recommended to set VARCHAR to 2083

File: DAO/MessageDAO.py (hunk 1 above)
Comment (ChristianPerez34) [reply DAO/MessageDAO.py]: Could you run both queries separately and post pictures of tables? What if you switch the position of num and hid?

Comment (javierbustillo) [reply DAO/MessageDAO.py]: ... Se me habia quedado el natural INNER join. In my defense I was using sql server yesterday

Comment (ChristianPerez34) [reply DAO/MessageDAO.py]: It happens

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]
Must resolve discussion

File: DAO/ChatDAO.py [hunk 4]
```diff
@@ -12,9 +12,10 @@ def get_chat_messages(self, cid):
                 "from messages left outer join " \
                 "like_count on messages.mid = like_count.mid left outer join photo on " \
                 "messages.mid = photo.mid left outer join dislike_count on messages.mid = dislike_count.mid " \
-                "left outer join users on messages.uid = users.uid where messages.cid = %s"
+                "left outer join users on messages.uid = users.uid where messages.cid = %s " \
+                "ORDER BY messages.created_on DESC"
```
Comment (ChristianPerez34) [inline DAO/ChatDAO.py:16]: Is this order from most recent message to oldest?

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]
Question about message ordering

File: DAO/ChatDAO.py (hunk 4 above)
Comment (javierbustillo) [reply DAO/ChatDAO.py:16]: Yes

**Pull request merged**